src/lang.rs

```diff
@@ -274,6 +278,9 @@ impl ShadowLang {
                 return Err(err);
             };
         }
+        if let Ok(dir) = original_path {
+            let _ = env::set_current_dir(dir);
```

Comment:
Can do a similar thing here:
```suggestion
            assert!(env::set_current_dir(dir).is_ok());
```